internal/services/hybridcompute/arc_machine_data_source.go

```diff
@@ -116,23 +116,23 @@ type ErrorAdditionalInfoModel struct {
 	Type string `tfschema:"type"`
 }
 
-type HybridComputeMachineDataSource struct{}
+type ArcMachineDataSource struct{}
 
-var _ sdk.DataSource = HybridComputeMachineDataSource{}
+var _ sdk.DataSource = ArcMachineDataSource{}
 
-func (r HybridComputeMachineDataSource) ResourceType() string {
-	return "azurerm_hybrid_compute_machine"
+func (r ArcMachineDataSource) ResourceType() string {
+	return "azurerm_arc_machine"
```

Comment:
This is a breaking change, we'd need to add a new data source with the correct name and deprecate this one.

Reply:
Agreed, I will add this to a sperate data source.